### PR TITLE
[Update] Apply changes to APIs after credix deployment

### PR DESCRIPTION
**Pull request by uxd-vincent**
Updating dependencies following credix devnet and mainnet latest deployments

### CODE REVIEW

File: programs/uxd/src/instructions/credix_lp/rebalance_request_execute_from_credix_lp_depository.rs [hunk 1]
```diff
@@ -227,25 +227,14 @@ pub(crate) fn handler(ctx: Context<RebalanceRequestExecuteFromCredixLpDepository
...
-        require!(
-            (start_of_redeem_phase_timestamp..end_of_redeem_phase_timestamp)
-                .contains(&Clock::get()?.unix_timestamp),
-            UxdError::InvalidCredixWithdrawEpochRequestPeriod,
-        );
-    }
+    require!(
+        ctx.accounts.credix_withdraw_epoch.status()?
+            == credix_client::WithdrawEpochStatus::RedeemPhase,
+        UxdError::InvalidCredixWithdrawEpochRedeemPhase,
```
Comment (uxd-vincent) [inline programs/uxd/src/instructions/credix_lp/rebalance_request_execute_from_credix_lp_depository.rs:236]: this should fix any possible typo in the timestamp computation code

File: programs/uxd/src/instructions/credix_lp/rebalance_request_create_from_credix_lp_depository.rs [hunk 2]
```diff
@@ -148,20 +148,14 @@ pub(crate) fn handler(ctx: Context<RebalanceRequestCreateFromCredixLpDepository>
...
-        require!(
-            (start_of_request_phase_timestamp..end_of_request_phase_timestamp)
-                .contains(&Clock::get()?.unix_timestamp),
-            UxdError::InvalidCredixWithdrawEpochRequestPeriod,
-        );
-    }
+    require!(
+        ctx.accounts.credix_withdraw_epoch.status()?
+            == credix_client::WithdrawEpochStatus::RequestPhase,
+        UxdError::InvalidCredixWithdrawEpochRequestPhase,
```
Comment (uxd-vincent) [inline programs/uxd/src/instructions/credix_lp/rebalance_request_create_from_credix_lp_depository.rs:157]: this should fix any possible typo in the timestamp computation code

File: programs/uxd/Cargo.toml [hunk 3]
```diff
@@ -36,13 +36,16 @@ mercurial-vault = { git = "https://github.com/mercurial-finance/vault-sdk", feat
...
+], default-features = false }
 
 [dev-dependencies]
 proptest = "1.0.0"
-solana-program = "1.15.2"
-solana-program-test = "1.15.2"
-solana-sdk = "1.15.2"
+solana-program = "=1.14.17"
+solana-program-test = "=1.14.17"
+solana-sdk = "=1.14.17"
```
Comment (uxd-vincent) [inline programs/uxd/Cargo.toml:48]: match main net version, out dependencies dont work on 1.16.0 yet sadness (mercurial)

**Review by acamill — APPROVED** [review]
lgtm

**Pull request merged**